### PR TITLE
Use twine for PyPI uploads.

### PR DESCRIPTION
**Pull request by jezdez**
Fix for https://github.com/travis-ci/dpl/issues/253

### CODE REVIEW

**Comment by alex** [comment]
looks reasonable to me!

File: lib/dpl/provider/pypi.rb [hunk 1]
```diff
@@ -64,7 +65,9 @@ def needs_key?
 
       def push_app
         context.shell "python setup.py register -r #{options[:server] || 'pypi'}"
```
Comment (dstufft) [inline lib/dpl/provider/pypi.rb]: You might want to remove this line? Unless the project in question has never been registered with PyPI *ever* in it's entire history this line is not important and will negate any security benefits of using twine.

**Comment by dstufft** [comment]
One comment, otherwise looks good to me.

File: lib/dpl/provider/pypi.rb (hunk 1 above)
Comment (dstufft) [inline lib/dpl/provider/pypi.rb]: I should mention too, that Warehouse removes even the "you must have ran ``setup.py register`` at least once ever" caveat too.

Comment (jezdez) [inline lib/dpl/provider/pypi.rb]: I think that implies that the docs need to be updated that people have to register their project manually on the PyPI website to be safe?

Comment (jezdez) [inline lib/dpl/provider/pypi.rb]: The Travis docs, that is.

Comment (dstufft) [inline lib/dpl/provider/pypi.rb]: Yea, quite possibly. Not sure I ever read the docs but currently the only safe way is to do a one time register via the website (can just use version 0 and delete it) and then just use ``twine upload`` in the future.

**Pull request merged**